nifi-nar-bundles/nifi-lumberjack-bundle/nifi-lumberjack-processors/src/main/java/org/apache/nifi/processors/lumberjack/ListenLumberjack.java

```diff
@@ -0,0 +1,237 @@
+import org.apache.nifi.processor.util.listen.AbstractListenEventBatchingProcessor;
+import org.apache.nifi.processor.util.listen.dispatcher.AsyncChannelDispatcher;
+import org.apache.nifi.processor.util.listen.dispatcher.ChannelDispatcher;
+import org.apache.nifi.processor.util.listen.dispatcher.SocketChannelDispatcher;
+import org.apache.nifi.processor.util.listen.event.EventFactory;
+import org.apache.nifi.processor.util.listen.handler.ChannelHandlerFactory;
+import org.apache.nifi.processor.util.listen.response.ChannelResponder;
+import org.apache.nifi.processor.util.listen.response.ChannelResponse;
+import org.apache.nifi.processors.lumberjack.event.LumberjackEvent;
+import org.apache.nifi.processors.lumberjack.event.LumberjackEventFactory;
+import org.apache.nifi.processors.lumberjack.frame.LumberjackEncoder;
+import org.apache.nifi.processors.lumberjack.handler.LumberjackSocketChannelHandlerFactory;
+import org.apache.nifi.processors.lumberjack.response.LumberjackChannelResponse;
+import org.apache.nifi.processors.lumberjack.response.LumberjackResponse;
+import org.apache.nifi.ssl.SSLContextService;
+
+import javax.net.ssl.SSLContext;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Collection;
+import java.util.ArrayList;
+import java.util.concurrent.BlockingQueue;
+
+
+@InputRequirement(InputRequirement.Requirement.INPUT_FORBIDDEN)
+@Tags({"listen", "lumberjack", "tcp", "logs"})
+@CapabilityDescription("Listens for Lumberjack messages being sent to a given port over TCP. Each message will be " +
+        "acknowledged after successfully writing the message to a FlowFile. Each FlowFile will contain data " +
+        "portion of one or more Lumberjack frames. In the case where the Lumberjack frames contain syslog messages, the " +
+        "output of this processor can be sent to a ParseSyslog processor for further processing.")
+@WritesAttributes({
+        @WritesAttribute(attribute="lumberjack.sender", description="The sending host of the messages."),
+        @WritesAttribute(attribute="lumberjack.port", description="The sending port the messages were received over."),
+        @WritesAttribute(attribute="lumberjack.sequencenumber", description="The sequence number of the message. Only included if <Batch Size> is 1."),
+        @WritesAttribute(attribute="lumberjack.*", description="The keys and respective values as sent by the lumberjack producer. Only included if <Batch Size> is 1."),
+        @WritesAttribute(attribute="mime.type", description="The mime.type of the content which is text/plain")
+})
+@SeeAlso(classNames = {"org.apache.nifi.processors.standard.ParseSyslog"})
+public class ListenLumberjack extends AbstractListenEventBatchingProcessor<LumberjackEvent> {
+
+    public static final PropertyDescriptor SSL_CONTEXT_SERVICE = new PropertyDescriptor.Builder()
```

Comment:
Are you referring making the SSL required?

If so, the reason why it is not required is that Lumberjack itself does not mandate TLS (although nearly every implementation does). The situation get a bit worse when we decide to support Lumberjack v2 as beats (lumberjack-forwarder successor) implementation treats TLS as optional.

So we can chose between:

1. Make it required but potentially change it back to optional once we decide to move into v2 support 
2. Make it optional and leave to the user to chose.
 
Let me know your preferred option and I will adjust.